application/helpers/expressions/em_manager_helper.php

```diff
@@ -9972,7 +9935,34 @@ static public function ShowSurveyLogicFile($sid, $gid=NULL, $qid=NULL,$LEMdebugL
                 // FINALLY, SHOW THE QUESTION ROW(S), COLOR-CODING QUESTIONS THAT CONTAIN ERRORS
                 //////
                 $errclass = ($errorCount > 0) ? 'danger': '';
-                $errText=($errorCount > 0) ? "<br><em class='label label-danger'>".$LEM->ngT("This question has at least {n} error.|This question has at least {n} errors.",$errorCount)."<em>" : "";
+                $errText=($errorCount > 0) ? "<br><em class='label label-danger'>".$LEM->ngT("This question has at least {n} error.|This question has at least {n} errors.",$errorCount)."</em>" : "";
+                /* Construct the warnings */
+                $sWarningsText = "";
+                if(count($aWarnings) > 0) {
+                    $sWarningsText .= "<div class='alert alert-warning'>";
+                    $sWarningsText .= "<strong class=''>".$LEM->ngT("This question has at least {n} warning.|This question has at least {n} warnings.",count($aWarnings))."</strong>";
+                    $sWarningsText .= "<ul class='list-unstyled small text-warning'>";
+                    $warningsDone = array();
+                    foreach($aWarnings as $aWarning) {
```

Comment:
I'm not too happy about this piece of code. I you'd done OOP, this loop would only contain `$warning->__toString()` and not HTML. And RDP_Warnings could be a Collection object, also using the __toString magic method.

Reply:
But I think I'll merge it today anyway, and release. We can add some @todo and maybe fix it later.

Reply:
Yes, sure … but all of this function already do a complete HTML …